src/core_modules/capture-core/HOC/withD2InContext.js

```diff
@@ -3,22 +3,23 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import getD2 from '../d2/d2Instance';
 
-export default () => (InnerComponent: React.ComponentType<any>) => class D2ContextAdder extends React.Component<any> {
-    static childContextTypes = {
-        d2: PropTypes.object,
-    };
-
-    getChildContext() {
-        return {
-            d2: getD2(),
+export const withD2InContext =
+    () => (InnerComponent: React.ComponentType<any>) => class D2ContextAdder extends React.Component<any> {
+        static childContextTypes = {
+            d2: PropTypes.object,
         };
-    }
 
-    render() {
-        return (
-            <InnerComponent
-                {...this.props}
-            />
-        );
-    }
-};
+        getChildContext() {
+            return {
+                d2: getD2(),
+            };
+        }
+
+        render() {
+            return (
+                <InnerComponent
+                    {...this.props}
+                />
+            );
+        }
+    };
```

Comment:
This could be done in a different PR, but we have an adapter library for `d2` access in the platform:

https://github.com/dhis2/app-runtime-adapter-d2

Reply:
Aa, thanks, was not familiar with this one. Will leave this for a different PR though.